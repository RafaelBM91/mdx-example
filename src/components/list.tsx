import * as  React from 'react';
import styled from 'styled-components';
import { ListContext } from '@app/store/context';
import * as actionTypes from '@app/store/actionTypes';

interface ItemModel {
    name: string;
    age: number;
}

const Field = styled.label`
    display: flex;
    flex-direction: column;
    width: fit-content;
    margin: 10px 0 0 0;
`;

export const List: React.FunctionComponent<{}> = () => {
    const { dispatch, state: { items } } = React.useContext(ListContext);
    let name: HTMLInputElement | null = null;
    let age: HTMLInputElement | null = null;

    const AddItem = () => {
        if (name && age) {
            if (name.value.length && age.value.length) {
                dispatch ({
                    type: actionTypes.ADD_ITEM,
                    payload: { name: name.value, age: parseInt(age.value) }
                });
                name.value = '';
                age.value = '';
            }
        }
    };

    return (
        <div>
            <div id="coverList">
                <ul>
                    {
                        items.map((item, index) => (
                            <li key={index}>
                                <b>{item.name}</b>:&nbsp;
                                {item.age}
                            </li>
                        ))
                    }
                </ul>
            </div>
            <div>
                <Field>
                    Name:
                    <input type="text" name="name" ref={(ref) => name = ref} />
                </Field>
                <Field>
                    Age:
                    <input type="number" name="age" ref={(ref) => age = ref} />
                </Field>
                <Field>
                    <button onClick={AddItem}>Add Item</button>
                </Field>
            </div>
        </div>
    );
};
