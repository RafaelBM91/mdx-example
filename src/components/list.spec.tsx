import * as React from "react";
import * as renderer from "react-test-renderer";
import { List } from './list';
import {shallow} from "enzyme";

describe('Test List Component', () => {
    it('Snapshot Component', () => {
        const list = renderer
            .create(<List />)
            .   toJSON();
        expect(list).toMatchSnapshot();
    });
    it('Functional Component', () => {
        const list = shallow(<List />);
        let inputName = list.find('input[name="name"]');
        let inputAge = list.find('input[name="age"]');
        let button = list.find('button');
        let items = list.find('div#coverList > ul > li');
        expect(inputName.length).toEqual(1);
        expect(inputAge.length).toEqual(1);
        expect(button.length).toEqual(1);
        expect(items.length).toEqual(0);
    });
});
