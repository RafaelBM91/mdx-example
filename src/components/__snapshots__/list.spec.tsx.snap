// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test List Component Snapshot Component 1`] = `
<div>
  <div
    id="coverList"
  >
    <ul />
  </div>
  <div>
    <label
      className="sc-bdVaJa cxriQp"
    >
      Name:
      <input
        name="name"
        type="text"
      />
    </label>
    <label
      className="sc-bdVaJa cxriQp"
    >
      Age:
      <input
        name="age"
        type="number"
      />
    </label>
    <label
      className="sc-bdVaJa cxriQp"
    >
      <button
        onClick={[Function]}
      >
        Add Item
      </button>
    </label>
  </div>
</div>
`;
